### PR TITLE
DEVPROD-4045 test log redactor

### PR DESCRIPTION
**Pull request by ybrill**
[DEVPROD-4045](https://jira.mongodb.org/browse/DEVPROD-4045)

### Description
Wrap the test senders with the redacting sender. If something looks odd it's probably because doing it the straightforward way caused an import cycle 😮‍💨 .

### Testing
In staging, I created a private variable, `jonathan_test: RUN`. This caused the word RUN to get redacted from our go tests, such as [here](https://parsley-staging.corp.mongodb.com/test/evg_ubuntu2204_test_agent_patch_9f7bd84c73f9d7bd2343f77a55fc4171e780b214_65f30ac534e4520007691452_24_03_14_14_33_50/1/88116fd731f7fea1f714e94ca92f4ceb?bookmarks=0,564&shareLine=12).

### CODE REVIEW

File: agent/command/results_utils.go [hunk 1]
```diff
@@ -151,3 +152,15 @@ func makeCedarTestResults(id string, t *task.Task, results []testresult.TestResu
 
 	return rs, failed
 }
+
+// ToRedact returns the full list of expansion keys whose values
+// should get redacted from task logs.
+func ToRedact(redacted map[string]bool) []string {
```
Comment (julianedwards) [inline agent/command/results_utils.go:158]: Can this get moved to `agent/internal/redactor`?

File: agent/internal/redactor/redacting_sender.go [hunk 2]
```diff
@@ -19,6 +19,13 @@ type redactingSender struct {
 	send.Sender
 }
 
+type RedactionOptions struct {
```
Comment (julianedwards) [inline agent/internal/redactor/redacting_sender.go:22]: Nit: add documentation here.

**Review by julianedwards — CHANGES_REQUESTED** [review]
This is a really elegant solution!

File: agent/command/results_utils.go (hunk 1 above)
Comment (ybrill) [reply agent/command/results_utils.go]: This was one of the things that had been causing an import cycle.
I rearranged things so it wouldn't.

**Comment by ybrill** [comment]
> This is a really elegant solution!

Thank you!

File: agent/internal/redactor/redacting_sender.go (hunk 2 above)
Comment (ybrill) [reply agent/internal/redactor/redacting_sender.go:24]: done

**Review by julianedwards — APPROVED** [review]

**Pull request merged**